python/langsmith/run_helpers.py

```diff
@@ -970,14 +938,18 @@ def _collect_extra(extra_outer: dict, langsmith_extra: LangSmithExtra) -> dict:
     return extra_inner
 
 
-def _get_parent_run(langsmith_extra: LangSmithExtra) -> Optional[run_trees.RunTree]:
+def _get_parent_run(
+    langsmith_extra: LangSmithExtra, config: Optional[dict] = None
+) -> Optional[run_trees.RunTree]:
     parent = langsmith_extra.get("parent")
     if isinstance(parent, run_trees.RunTree):
         return parent
     if isinstance(parent, dict):
         return run_trees.RunTree.from_headers(parent)
     if isinstance(parent, str):
         return run_trees.RunTree.from_dotted_order(parent)
+    if rt := run_trees.RunTree.from_runnable_config(config):
```

Comment:
Wouldn't this raise an import error if the user doesn't have langchain installed? We should probably:

1. catch that error
2. maybe cache whether or not it's installed to avoid repeatedly trying to import

Reply:
I think this should also be lower precedence than the "run-tree" condition in 953